plugins/search-backend-node/src/IndexBuilder.ts

```diff
@@ -100,28 +100,31 @@ export class IndexBuilder {
   async build() {
     return Promise.all(
       Object.keys(this.collators).map(async type => {
-        const decorators: DocumentDecorator[] = (
-          this.decorators['*'] || []
-        ).concat(this.decorators[type] || []);
+        setInterval(async () => {
+          const decorators: DocumentDecorator[] = (
+            this.decorators['*'] || []
+          ).concat(this.decorators[type] || []);
 
-        this.logger.info(
-          `Collating documents for ${type} via ${this.collators[type].collate.constructor.name}`,
-        );
-        let documents = await this.collators[type].collate.execute();
-        for (let i = 0; i < decorators.length; i++) {
           this.logger.info(
-            `Decorating ${type} documents via ${decorators[i].constructor.name}`,
+            `Collating documents for ${type} via ${this.collators[type].collate.constructor.name}`,
           );
-          documents = await decorators[i].execute(documents);
-        }
+          let documents = await this.collators[type].collate.execute();
+          for (let i = 0; i < decorators.length; i++) {
+            this.logger.info(
+              `Decorating ${type} documents via ${decorators[i].constructor.name}`,
+            );
+            documents = await decorators[i].execute(documents);
+          }
 
-        if (!documents || documents.length === 0) {
-          this.logger.info(`No documents for type "${type}" to index`);
-          return;
-        }
+          if (!documents || documents.length === 0) {
+            this.logger.info(`No documents for type "${type}" to index`);
```

Comment:
logging question on this one also: is this rare? or does this jus mean in this refresh loop, nothing has changed TO index since the last?

if former (rare), maybe this should be WARN, but if latter, then maybe this should be debug? (again if say backstage sits idle for 12 hours overnight, but the search refresh runs every 15 mins, this would print out 48 times in my logs ?)

Reply:
What it means is really that a collator is registered but no documents were collected to be indexed. So probably it is rare, depending on how many collators you have registered. But even if it is rare I think it can be a debug one, because you are only interested in this in debug mode I think 🤔